compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -302,6 +368,15 @@ LogicalResult coreToAIE(IRRewriter &rewriter, AMDAIE::CoreOp coreOp,
             .Case<linalg::LinalgOp>([&](auto linalgOp) {
               return coreLinalgOpToAIE(rewriter, linalgOp, mapper, toBeErased);
             })
+            .Case<memref::ExtractStridedMetadataOp>(
```

Comment:
Why does the default implementation not work? That one should remap operands?

Reply:
Because the `remapOpOperands` doesn't seem to be reflecting the necessary change on the inferred types.

Eg:
```
memref.extract_strided_metadata %1 : memref<32x32xi32, 2> -> memref<i32, 2>, index, index, index, index, index
```
In the above case `remapOpOperands` would change it to `memref<32x32xi32, 1> -> memref<i32, 2>, index, index, index, index, index` (Note the memory space mismatch).

It should indeed be `memref<32x32xi32, 1> -> memref<i32, 1>, index, index, index, index, index` which can only happen if I create a new `memref.extract_strided_metadata` op.

Reply:
I see, yeah, this might become more clean when we refactor more of the lower-level passes, so that memory spaces don't have to change for example.